Payments

Dogober opened the pull request:
- Create payment model 
- Add paypal app credentials to .env

Dogober committed:
change command name inside README.MD from requrements -> requirements

Dogober committed:
create Payment model

MindDevastation commented on src/payments/models.py:
```diff
@@ -1,3 +1,25 @@
+    )
+    session_url = models.URLField()
+    session_id = models.IntegerField()
+    models.DecimalField(max_digits=5, decimal_places=2)
```
Assign a variable here and add positive=True parameter

MindDevastation commented on src/payments/models.py:
```diff
@@ -1,3 +1,25 @@
+        Borrowing, on_delete=models.CASCADE, related_name="payments"
+    )
+    session_url = models.URLField()
+    session_id = models.IntegerField()
```
Use the UUIDField model for session_id

MindDevastation reviewed (changes requested):
Here's some requirements needed:

1. Assign a variable to models.DecimalField(max_digits=5, decimal_places=2) and add the positive=True parameter. Amount can`t be negative
2. Add borrow_date as a field in the model and add a save method to set it from borrowings.borrow_date.
3. Use the UUIDField model for session_id (default=uuid4, editable=False, unique=True).
4. Add created_at and updated_at fields with auto-filling.
5. Add the __str__ method.
6. Add indexes for the fields "status" and "borrowing".
7. Add additional statuses (FAILED, CANCELED, REFUNDED).
8. Add addition choise field for currency (optional)

Dogober committed:
Update Payment model:
- Added amount field with positive=True to ensure non-negative payments.
- Added borrow_date field and modified save method to set it from borrowings.borrow_date.
- Changed session_id to UUIDField with default=uuid4 and unique constraint.
- Added created_at and updated_at fields with auto-filling.
- Implemented __str__ method for better string representation.
- Added indexes for fields 'status' and 'borrowing' to improve query performance.
- Expanded status choices to include FAILED, CANCELED, and REFUNDED.
- Added optional currency field with choices.

MindDevastation commented on src/payments/models.py:
```diff
@@ -1,3 +1,56 @@
+        FINE = "FINE", "Penalty"
+
+    class Currency(models.TextChoices):
+        UAH = "UAH", "Ukrainian hryvnia"
```
Remove uah. PayPal is not support this currency

MindDevastation reviewed (changes requested):
Remove uah. PayPal is not support this currency

Dogober committed:
delete UAH currency due to incompatibility with PayPal

MindDevastation commented on src/payments/models.py:
```diff
@@ -1,3 +1,55 @@
+    money_to_pay = models.DecimalField(max_digits=5, decimal_places=2, positive=True)
+    borrow_date = models.DateField()
+    currency = models.CharField(
+        max_length=3, choices=Currency.choices, default=Currency.UAH, blank=True
```
change to USD

MindDevastation reviewed (changes requested):
change default currncy to usd

Dogober committed:
Update Payment model to set default currency from UAH to USD.
